client/lists/carousel.tsx

```diff
@@ -0,0 +1,190 @@
+  | 'hasPrevData'
+  | 'onLoadPrevData'
+  | 'nextLoadingEnabled'
+  | 'children'
+>
+
+interface CarouselProps {
+  children: React.ReactNode
+  infiniteListProps?: CarouselInfiniteListProps
+  className?: string
+}
+
+/**
+ * A component which utilizes InfiniteScrollList to display items in a horizontal list with built-in
+ * scrolling. It only supports dynamically adding items at the end of the list.
+ */
+export function Carousel({ children, infiniteListProps = {}, className }: CarouselProps) {
+  const { t } = useTranslation()
+
+  const [translateWidth, setTranslateWidth] = useState(0)
+  const [carouselWidth, setCarouselWidth] = useState(0)
+  const [contentWidth, setContentWidth] = useState(0)
+  const [stepWidth, setStepWidth] = useState(0)
+  const [hasPrevItems, setHasPrevItems] = useState(false)
+  const [hasNextItems, setHasNextItems] = useState(false)
+
+  const carouselRef = useRef<HTMLDivElement>(null)
+  const contentRef = useRef<HTMLDivElement>(null)
+  const animationIdRef = useRef<number | null>(null)
+
+  const calcCarouselWidth = useCallback(() => {
+    if (!carouselRef.current || !contentRef.current) {
+      return
+    }
+
+    // This is the width of currently visible items
+    let carouselWidth = carouselRef.current.getBoundingClientRect().width
+    // This is the width of all items in total
+    const contentWidth = contentRef.current.getBoundingClientRect().width
+
+    const hasPrevItems = translateWidth < 0
+    const adjustment = hasPrevItems ? BUTTON_WIDTH : 0
+    const hasNextItems = contentWidth - carouselWidth + adjustment > Math.abs(translateWidth)
+
+    if (hasPrevItems && hasNextItems) {
+      carouselWidth = carouselWidth - BUTTON_WIDTH * 2 // both prev and next buttons are visible
+    } else if (hasPrevItems || hasNextItems) {
+      carouselWidth = carouselWidth - BUTTON_WIDTH // only one of the buttons is visible
+    }
+
+    // Scroll 60% of the currently visible items
+    const stepWidth = carouselWidth * 0.6
+
+    setCarouselWidth(carouselWidth)
+    setContentWidth(contentWidth)
+    setStepWidth(stepWidth)
+    setHasPrevItems(hasPrevItems)
+    setHasNextItems(hasNextItems)
+  }, [translateWidth])
+
+  useEffect(() => {
+    calcCarouselWidth()
+    // Need to re-calculate carousel width when children changes in case they were dynamically added
+  }, [calcCarouselWidth, children])
+
+  useEffect(() => {
+    return () => {
+      if (animationIdRef.current !== null) {
+        window.cancelAnimationFrame(animationIdRef.current)
+      }
+    }
+  }, [])
+
+  const animatePosition = (delta: number) => {
+    setTranslateWidth(translateWidth => translateWidth + delta)
+  }
+
+  const contentStyle = { transform: `translateX(${translateWidth}px)` }
+  const showPrevButton = hasPrevItems
+  const showNextButton = hasNextItems && !infiniteListProps.isLoadingNext
+
+  return (
+    <CarouselContainer ref={carouselRef}>
+      <WindowListener event='resize' listener={calcCarouselWidth} />
```

Comment:
We should probably move to a resizeobserver instead of this (I know this PR isn't to rewrite this component so it doesn't need to be part of this, but if we ever plan to use this going forward we should fix it up)